prusti-viper/src/encoder/pure_function_encoder.rs

```diff
@@ -151,26 +151,9 @@ impl<'p, 'v: 'p, 'tcx: 'v> PureFunctionEncoder<'p, 'v, 'tcx> {
             );
         }
 
-        // TODO: Clean up code duplication:
-        //let contract = self.encoder.get_procedure_contract_for_def(self.proc_def_id);
-        let contract = {
-            let opt_fun_spec = self.encoder.get_procedure_specs(self.proc_def_id);
-            let fun_spec = match opt_fun_spec {
-                Some(fun_spec) => fun_spec.clone(),
-                None => {
-                    debug!("Procedure {:?} has no specification", self.proc_def_id);
-                    typed::SpecificationSet::Procedure(typed::ProcedureSpecification::empty())
-                }
-            };
-            let tymap = self.encoder.current_tymap();
-            let contract = compute_procedure_contract(
-                self.proc_def_id,
-                self.encoder.env().tcx(),
-                fun_spec,
-                Some(&tymap),
```

Comment:
As far as I can tell, this line is the only difference between the code in `encoder` and here. In `encoder`, the last argument is `None`, here the `tymap` is used. I am not sure what this implies, but I don't think it affected the test results at all.

Reply:
From the implementation of `compute_procedure_contract` the `tymap` seems to be used for pure functions with type parameters; but I don't have an example to demonstrate it.